frontends/p4/sideEffects.cpp

```diff
@@ -600,6 +608,8 @@ const IR::Node* DoSimplifyExpressions::postorder(IR::AssignmentStatement* statem
     auto right = parts->final;
     CHECK_NULL(right);
     parts->statements->push_back(new IR::AssignmentStatement(statement->srcInfo, left, right));
+    if (parts->statements->size() == 1)
```

Comment:
these blocks will be eliminated later. Is there a good reason to avoid creating them altogether?

Reply:
Wasteful churn slowing down the compiler, making it harder to debug what is going on when trying to track where stuff is changing across all the passes of the compiler.